src/mem/ruby/structures/CacheMemory.cc

```diff
@@ -89,11 +86,21 @@ CacheMemory::CacheMemory(const Params &p)
 }
 
 void
-CacheMemory::init()
+CacheMemory::setRubySystem(RubySystem* rs)
 {
+    dataArray.setClockPeriod(rs->clockPeriod());
+    tagArray.setClockPeriod(rs->clockPeriod());
+    atomicALUArray.setClockPeriod(rs->clockPeriod());
```

Comment:
Just a note: we could take this change and extend it to allow different caches to have different clocks.

This is not something we should do in this PR, but it is something we can now do more easily!